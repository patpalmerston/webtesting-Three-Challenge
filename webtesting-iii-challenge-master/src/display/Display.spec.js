import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
// import { toHaveClass } from '@testing-library/jest-dom'

import Display from './Display';

describe('Display SnapShot', () => {
	it('should match Display SnapShot', () => {
		const tree = renderer.create(<Display />);

		expect(tree.toJSON()).toMatchSnapshot();
	});
});

describe('closed prop', () => {
	it('displays closed if closed prop is === true', () => {
		const { getByText } = render(<Display closed={true} />);
		getByText('Closed');
	});

	it('displays open if closed prop is === false', () => {
		const { getByText } = render(<Display closed={false} />);
		getByText('Open');
	});
});

describe('locked prop', () => {
	it('displays Locked if the locked prop is true', () => {
		const { getByText } = render(<Display locked={true} />);
		getByText('Locked');
	});

	it('displays UnLocked if the locked props is false', () => {
		const { getByText } = render(<Display locked={false} />);
		getByText('Unlocked');
	});
});

describe('red-led class tests on true', () => {
	it('red-led on locked true', () => {
		const { container } = render(<Display locked={true} />);
		expect(container.querySelector('led red-led'));
	});

	it('red-led on closed true', () => {
		const { container } = render(<Display closed={true} />);
		expect(container.querySelector('led red-led'));
	});
});


