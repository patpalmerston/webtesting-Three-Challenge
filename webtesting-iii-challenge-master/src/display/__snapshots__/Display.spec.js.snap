// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashBoard SnapShot should match Display SnapShot 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;

exports[`Display SnapShot should match Display SnapShot 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;
